packages/expect/__typetests__/expect.test.ts

```diff
@@ -48,6 +57,7 @@ expectType<void>(
       expectType<Array<Error>>(this.suppressedErrors);
       expectType<string | undefined>(this.testPath);
       expectType<MatcherUtils>(this.utils);
+      expectType<Readonly<MatchersObject>>(this.matchers);
```

Comment:
This test should be written from usr perspective. Can be similar to documentation example:

<img width="623" alt="Screenshot 2022-12-21 at 20 16 59" src="https://user-images.githubusercontent.com/72159681/208975730-4d08261c-b337-4eda-8d9f-d660c17f3e62.png">

Could you take another look, please? Ideally `this.matchers.toMatchObject()` should not allow extra args just like `.toMatchObject()` matcher does. I understand that this could be complicated, but there are other type issues to fix.

Reply:
@JVBorges could you take a look at this feedback? We're getting really close 🙂

Reply:
Once again sorry for the delay. I'm a little busy right now, which is why I'm taking so long to reply.

Just a question to see if I got it right. I can keep passing the actual value to the corresponding built-in matcher next to the value to be compared, right? I just need to prevent the user from being able to pass more than these two or one arguments, for example: ```this.matchers.toBeTruthy(actual)``` or  ```this.matchers.toMatcheObject(actual, expectedObject)```

Also, calling it ```builtInMatcher``` is much better than ```internalMatcher```

Reply:
Yeah, it should (ideally) be a type error. If it proves hard in practice we can probably punt on it